### PR TITLE
#158755849 Static page management

### PR DESCRIPTION
**Pull request by OlatunjiYso**
#### What does this PR do?
This PR allows the admin to manage static pages and users view static pages
#### Description of Task to be completed?
- add model for storing static pages in database
- create plugin for static pages
- add templates for viewing static page (users)
- add templates for static page management (admin)
- add components and containers for static pages
- add forms to add and edit static page
- add table for managing static pages
#### How should this be manually tested?
- pull the branch
- run `reaction reset -n`
- reaction to start the app
- signin as admin and find the pages icon on the admin side-nav
- select it and add and manage pages
#### Screenshots (if appropriate)
screen shot 2018-07-19 at 12 28 41 pm
screen shot 2018-07-19 at 12 24 49 pm
screen shot 2018-07-19 at 12 27 17 pm
screen shot 2018-07-19 at 12 26 47 pm

#### What are the relevant pivotal tracker stories?
[Finishes #158755849](https://www.pivotaltracker.com/n/projects/2182264)

#### Any background context you want to provide?
N/A

### CODE REVIEW

**Comment by ddouglasz** [comment]
LGTM 😋. At least I came around to your desk to see what you were doing. Good job bro.

**Comment by saheedt** [comment]
@Daramola98 and I tested this, look good to go. Great job @OlatunjiYso

**Comment by abdulfataiaka** [comment]
@OlatunjiYso, Is there a reason why you are changing core plugins instead of creating a new plugin for static pages?

**Comment by OlatunjiYso** [comment]
@abdulfataiaka , I created my react components within the core plugin because I found other other react components there.

**Comment by Rhotimee** [comment]
Works fine. LGTM

**Comment by abdulfataiaka** [comment]
@OlatunjiYso, your `staticPage` component file was placed in the `accounts` plugin folder, which was meant to handle `account features` only and does not have anything to do with static pages, the plugin for the static page feature should be in the `/imports/plugins/custom` folder same way the `vendor-tour` plugin created by @Daramola98 is, and plugins are not in the core folder because they have react components, its because they introduce core feature into `reactioncommerce`.

**Comment by abdulfataiaka** [comment]
@OlatunjiYso 

- The scope of your commit message should be pascal cased and not spaced
- You commit message should have a very short, precise and descriptive title
- The commit message body should hold the listing of what have been done to implement the feature preceded with -the `-` character

Please checkout the link below
https://github.com/andela/asgard-rc/wiki/Git-naming-conventions-and-best-practices

**Comment by abdulfataiaka** [comment]
@OlatunjiYso, the storyId in your PR title should not be in this => `[    ]`, and the first letter of your PR title should be capitalized as in a `sentence`

**Comment by OlatunjiYso** [comment]
@abdulfataiaka , I have implemented the feedback given.

**Comment by abdulfataiaka** [comment]
@OlatunjiYso, please pay attention to the convections, you seem to be missing out on some things, in your branch name
`ft-static-pages-#158755849` should be `ft-static-pages-158755849`
And in your commit message
should start with `feat` not `ft`

**Comment by abdulfataiaka** [comment]
@OlatunjiYso, notice that plugins folder names are not `camel` or `pascal` cased, please rectify this.

**Comment by abdulfataiaka** [comment]
@OlatunjiYso, lastly, why do we have three separate plugins for same feature, why can't we have every functionalities of the static page scoped under on plugin folder?

File: imports/plugins/included/default-theme/client/styles/staticPages.less [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    line-height: 30px;
+}
+
+.staticHead {
+    font-size: 36px;
+}
+
+.pageManager {
+    margin-top: 4rem;
+}
```
Comment (clintfidel) [inline imports/plugins/included/default-theme/client/styles/staticPages.less:12]: @OlatunjiYso , I think you should look into the naming convention for styles, you using pascalCase and snake-case, follow a convention a use it all through.

**Review by clintfidel — COMMENTED** [review]
Good one @OlatunjiYso , please look into the comment I dropped, thanks

**Comment by clintfidel** [comment]
@OlatunjiYso , I would suggest that you follow color scheme while designing the template for static pages as the colors you used seems off, please look into the color scheme. Thanks

**Comment by OlatunjiYso** [comment]
@abdulfataiaka , the names of the static page folders are only similar, they are very much different.
As such, I may not be able to scope them into a single folder. 

- The first folder `static-page` is for viewing the static page
- The second folder `static-page-management` is for the admin who adds, edit and delete static pages
- The last folder `static-page-component` is for the nav.

Putting all these together will require me declare a common template, route and name for the three which I believe wont work out.
What do you think about this?

**Comment by abdulfataiaka** [comment]
@OlatunjiYso, you have made a good point, so please address @clintfidel's comments.

**Comment by abdulfataiaka** [comment]
@OlatunjiYso, this is what I see when I tried it out 👇 

screen shot 2018-07-19 at 2 38 42 pm
screen shot 2018-07-19 at 2 39 14 pm

**Comment by abdulfataiaka** [comment]
@OlatunjiYso, you seem to have not addressed the branch name and commit message comments

**Comment by OlatunjiYso** [comment]
@ddouglasz 
@saheedt 
@abdulfataiaka 
@clintfidel 
@Daramola98 
@Rhotimee 
@abdulfataiaka 
Please I may need to bring down this PR because,  implementing the TTL's feedback requires me to rename the branch.
I have renamed the branch and addresses other feedback too. I will be raising a new PR off that branch.
The PR will be up shortly, please bear with me.

**Comment by OlatunjiYso** [comment]
Please lets continue the conversation in the new PR

**Pull request closed without merging**